src/error/go.rs

```diff
@@ -65,3 +66,27 @@ impl fmt::Display for GoResult {
         }
     }
 }
+
+impl GoResult {
+    /// This is a wrapper around `impl TryFrom<GoResult> for Result<(), FfiError>` that uses a fallback
+    /// if output is not-empty, use that as the error message
+    /// otherwise, call default() to generate a default message
+    ///
+    /// Safety: this reads data from an externally provided buffer and assumes valid utf-8 encoding
+    /// Only call if you trust the code that provides output to be correct
+    pub unsafe fn into_ffi_result<F>(self, output: Buffer, default: F) -> Result<(), FfiError>
+    where
+        F: Fn() -> String,
+    {
+        self.try_into().unwrap_or_else(|_| {
+            let msg = if output.ptr.is_null() {
+                default()
+            } else {
+                // We initialize `output` with a null pointer. if it is not null,
+                // that means it was initialized by the go code, with values generated by `memory::allocate_rust`
+                String::from_utf8_lossy(&output.consume()).into()
```

Comment:
Why was this trailing `.into()` added? Isn't `msg` an owned `String` in both cases?

Reply:
Nope. `String::from_utf8_lossy()` takes `&String` and returns `Cow`.
Only discovered this when the compiler complained.

Reply:
Like it may return the original data, or it may return a new copy I guess.

Reply:
Ah, right, perfect. Zero-copy when nothing needs to be replaced.

Reply:
https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_lossy